sdk/nodejs/resource.ts

```diff
@@ -122,6 +122,10 @@ export abstract class Resource {
      * @param opts A bag of options that control this resource's behavior.
      */
     constructor(t: string, name: string, custom: boolean, props: Inputs = {}, opts: ResourceOptions = {}) {
+        if (opts.parent && !Resource.isInstance(opts.parent)) {
+            throw new Error(`Resource parent is not a valid Resource: ${opts.parent}`);
+        }
```

Comment:
note that thsi changes into an Error.  This should not have been a 'runerror'.  "RunError" means "do not print the stack, it's noisy".  In this case, we def *do* want to print the stack so the user can go and find the problem here.